src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1580,8 +1580,11 @@ public static class InternalTestHooks
         // Simulate 'System.Diagnostics.Stopwatch.IsHighResolution is false' to test Get-Uptime throw
         internal static bool StopwatchIsNotHighResolution;
 
+        // Will be either "delete" or "rename" during tests.
+        internal static string GciEnumerationAction = null;
+
         /// <summary>This member is used for internal test purposes.</summary>
-        public static void SetTestHook(string property, bool value)
+        public static void SetTestHook(string property, object value)
```

Comment:
I feel like it would be better to not allow both to be set at the same time, which the string property accomplishes.